app/services/payment_providers/adyen_service.rb

```diff
@@ -64,6 +64,23 @@ def handle_event(organization:, event_json:)
 
         result = service.preauthorise(organization, event)
         result.raise_if_error! || result
+      when 'REFUND'
+        service = CreditNotes::Refunds::AdyenService.new
+
+        provider_refund_id = event["pspReference"]
+        status = event["success"] == "true" ? :succeeded : :failed
+
+        result = service.update_status(provider_refund_id:, status:)
+        result.raise_if_error! || result
+      when 'REFUND_FAILED'
+        return result if event["success"] != "true"
+
+        service = CreditNotes::Refunds::AdyenService.new
+
+        provider_refund_id = event["pspReference"]
```

Comment:
Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols. (https://rubystyle.guide#consistent-string-literals)